lib/backend/helpers.go

```diff
@@ -61,3 +61,38 @@ func ReleaseLock(ctx context.Context, backend Backend, lockName string) error {
 	}
 	return nil
 }
+
+// ResetLockTTL resets the TTL on a given lock.
+func ResetLockTTL(ctx context.Context, backend Backend, lockName string) error {
+	if lockName == "" {
+		return trace.BadParameter("missing parameter lock name")
+	}
+
+	key := []byte(filepath.Join(locksPrefix, lockName))
+
+	item, err := backend.Get(ctx, key)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if _, err := backend.Put(ctx, *item); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
```

Comment:
Expiration is controlled by the `Item.Expires` field.  Performing a `Get + Put` does not change the value of `Item.Expires`, so this function won't do anything.  Also, `Put` doesn't guarantee that the lock wasn't expired/modified between the time `Get` was run and the time `Put` succeeds.

In fact, looking at how our locking system currently works, a "refresh" isn't really possible.  The reason for this is that we don't have a mechanism of distinguishing between "our" lock and a lock created by some other process.  I have an idea for how to fix this though:


Currently, locks are created like so:

```golang
backend.Create(ctx, Item{Key: key, Value: []byte{1}, Expires: backend.Clock().Now().UTC().Add(ttl)})
```

Notice that the value is always just a placeholder.  Lets change this to a random ID.  That way, processes will be able to distinguish their lock from other locks.

With the random ID strategy, a "refresh" of a lock would look something like this:

```golang
prev, err := backend.Get(ctx, key)
if err != nil {
    if trace.IsNotFound(err) {
        return trace.CompareFailed("cannot refresh lock %s (expired)")
    }
    return trace.Wrap(err)
}

if prev.Value != lockID {
    return trace.CompareFailed("cannot refresh lock %s (ownership changed)", lockID)
}

next := *prev
next.Expires = prev.Expires.Add(ttl)

_, err = backend.CompareAndSwap(ctx, *prev, next)
// ...
```

Reply:
I've now reworked how locking and refresh works. Have another look!